frontend/readhistory.lua

```diff
@@ -148,10 +135,12 @@ function ReadHistory:_readLegacyHistory()
             if path ~= nil and path ~= "" then
                 local file = DocSettings:getNameFromHistory(f)
                 if file ~= nil and file ~= "" then
-                    table.insert(
-                        self.hist,
-                        buildEntry(lfs.attributes(joinPath(history_dir, f), "modification"),
-                                   joinPath(path, file)))
+                    local item_path = joinPath(path, file)
+                    local item_time = lfs.attributes(joinPath(history_dir, f), "modification")
+                    local index = self:getIndexByTime(item_time)
+                    if self.hist[index].file ~= realpath(item_path) then
```

Comment:
Is that new/added test fixing some issue?

Reply:
Yes, previously items were just added, then sorted, and then duplicates were removed.
Now we check for existence and add legacy history items if they has not been added earlier.